cohorts/load.py

```diff
@@ -313,12 +313,42 @@ def load_from_cache(self, cache_name, patient_id, file_name):
         if not path.exists(cache_file):
             return None
 
+        df_provenance = self.generate_provenance(cache_name, patient_id)
+        df_provenance_previous = pd.read_csv(path.join(patient_cache_dir, "PROVENANCE"), index_col=None)
+
+        def provenance_set_from_df(df):
+            return set(dict(df.to_dict("split")["data"]).items())
+
+        def provenance_str_from_set(s):
+            return ["%s==%s" % (key, value) for (key, value) in s]
+
+        provenance_set = provenance_set_from_df(df_provenance)
+        provenance_set_previous = provenance_set_from_df(df_provenance_previous)
+        new_provenance_diff = provenance_set.difference(provenance_set_previous)
+        old_provenance_diff = provenance_set_previous.difference(provenance_set)
```

Comment:
Why do you need this two-way diff'ing procedure? Is it because we might have each provenance having additional/missing keys in the future?